Fix flakyness of the `KafkaConnectorIT` integration test

The `KafkaConnectorIT.testConnectorIsAutoRestarted` test seems to be flaky from time to time:

```
[ERROR] Failures: 
[ERROR] io.strimzi.operator.cluster.operator.assembly.KafkaConnectorIT.testConnectorIsAutoRestarted(VertxTestContext)
[ERROR]   Run 1: KafkaConnectorIT.testConnectorIsAutoRestarted(VertxTestContext) java.lang.AssertionError: 
Expected: is "RESTARTING"
     but: was "UNASSIGNED"
[ERROR]   Run 2: KafkaConnectorIT.testConnectorIsAutoRestarted(VertxTestContext) java.lang.AssertionError: 
Expected: is "RESTARTING"
     but: was "UNASSIGNED"
[ERROR]   Run 3: KafkaConnectorIT.testConnectorIsAutoRestarted(VertxTestContext) java.lang.AssertionError: 
Expected: is "RESTARTING"
     but: was "UNASSIGNED"
```

Full log can be found for example here: https://dev.azure.com/cncf/cbe844a9-ce48-479e-9b1b-3a18ed0139b7/_apis/build/builds/172416/logs/106

--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectorIT.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectorIT.java
@@ -65,6 +65,7 @@ import static org.hamcrest.Matchers.hasProperty;
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.oneOf;
 
 @ExtendWith(VertxExtension.class)
 public class KafkaConnectorIT {
@@ -406,7 +407,7 @@ public class KafkaConnectorIT {
                     return Future.succeededFuture();
                 } else {
                     Promise<Void> promise = Promise.promise();
-                    vertx.setTimer(2000, id -> operator.reconcileConnectorAndHandleResult(new Reconciliation("test", "KafkaConnect", namespace, "bogus"),
+                    vertx.setTimer(3000, id -> operator.reconcileConnectorAndHandleResult(new Reconciliation("test", "KafkaConnect", namespace, "bogus"),
                             "localhost", connectClient, true, connectorName, connector)
                         .onComplete(result -> promise.complete(result.result())));
                     return promise.future();
@@ -508,7 +509,8 @@ public class KafkaConnectorIT {
             assertThat(kafkaConnector.getStatus().getAutoRestart().getCount(), is(1));
             JsonObject connectorStatus = new JsonObject(kafkaConnector.getStatus().getConnectorStatus());
             assertThat(connectorStatus.getJsonObject("connector"), notNullValue());
-            assertThat(connectorStatus.getJsonObject("connector").getString("state"), is("RESTARTING"));
+            // Depending on the timing, the state might differ. So we try multiple different states => RESTARTING and UNASSIGNED.
+            assertThat(connectorStatus.getJsonObject("connector").getString("state"), is(oneOf("RESTARTING", "UNASSIGNED")));
             MetricsProvider metrics = new MicrometerMetricsProvider(BackendRegistries.getDefaultNow());
             MeterRegistry registry = metrics.meterRegistry();
             assertThat(registry.get(ConnectOperatorMetricsHolder.METRIC_AUTO_RESTARTS).tag("kind", KafkaConnector.RESOURCE_KIND).counter().count(), CoreMatchers.is(1.0));

